compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/loops/handlers/IndexedGetIterationHandlers.kt

```diff
@@ -75,7 +78,20 @@ internal class ArrayIterationHandler(context: CommonBackendContext) : IndexedGet
     private val supportsUnsignedArrays = context.optimizeLoopsOverUnsignedArrays
 
     override fun matchIterable(expression: IrExpression) =
-        expression.type.run { isArray() || isPrimitiveArray() || (supportsUnsignedArrays && isUnsignedArray()) }
+        expression.type.run { isArray() || isPrimitiveArray() || (supportsUnsignedArrays && isUnsignedArray()) } || expression.run {
+            this is IrCall && matcher(this)
+        }
+
+    private val matcher =
```

Comment:
Minor: give it a meaningful name - not just `matcher`, but something like `reversedArrayMatcher`.